### ISSUE TITLE
Possible bug, "fatal error: 'cairo/cairo.h' file not found"

### ISSUE THREAD

**Issue by viljamivirolainen** [giraffelibrary/giraffe#5]
I have followed the prerequisite and install instructions given in: 
http://giraffelibrary.org/sg-prerequisites.html
and 
http://giraffelibrary.org/sg-installation.html

I have a Macbook M1 Air with macOS Venture 13.0.1 (22A400). 

I have installed all dependencies with brew.

Here is output of ./configure:
`== Checking SML compilers ==
Checking MLton... NOT FOUND (using PATH)
Checking Poly/ML... found version 5.9 using pkg-config
  bin directory:     /opt/homebrew/Cellar/polyml/5.9/bin
  lib directory:     /opt/homebrew/Cellar/polyml/5.9/lib
  version:           5.9 >= 5.6

== Checking libraries ==
Checking atk-1.0... found version 2.38.0
Checking cairo-1.0... found version 1.16.0
Checking gdk-3.0... found version 3.24.36
Checking gdkpixbuf-2.0... found version 2.42.10
Checking gio-2.0... found version 2.74.4
Checking glib-2.0... found version 2.74.4
Checking gmodule-2.0... found version 2.74.4
Checking gobject-2.0... found version 2.74.4
Checking gtk-3.0... found version 3.24.36
Checking gtksource-3.0... found version 3.24.11
Checking harfbuzz-0.0... found version 6.0.0
Checking pango-1.0... found version 1.50.12
Checking pangocairo-1.0... found version 1.50.12
Checking vte-2.90... NOT FOUND
Checking vte-2.91... found version 0.70.2

Configuration summary

  Installation directory    /opt/giraffe

  Standard ML compilers   Using version
    MLton                   NOT AVAILABLE
    Poly/ML                 5.9

  Libraries               Using version           Giraffe Library API version
    atk-1.0                 2.38.0          >=      2.36.0
    cairo-1.0               1.16.0          >=      1.14.0
    gdk-3.0                 3.24.36         >=      3.24.24
    gdkpixbuf-2.0           2.42.10         >=      2.40.0
    gio-2.0                 2.74.4          >=      2.66.4
    glib-2.0                2.74.4          >=      2.66.4
    gmodule-2.0             2.74.4          >=      2.66.4
    gobject-2.0             2.74.4          >=      2.66.4
    gtk-3.0                 3.24.36         >=      3.24.24
    gtksource-3.0           3.24.11         >=      3.24.11
    harfbuzz-0.0            6.0.0           >=      2.7.2
    pango-1.0               1.50.12         >=      1.48.0
    pangocairo-1.0          1.50.12         >=      1.48.0
    vte-2.90                NOT AVAILABLE
    vte-2.91                0.70.2          >=      0.62.1`

Here is the output of make:
```
ln -sf POLYML_F_F_I-sig.sml src/ffi/polyml/POLYML_F_F_I-sig.sml
ln -sf PolyMLFFI.sml src/ffi/polyml/PolyMLFFI.sml
ln -sf PolyMLFFIIntConvs-IntInfAsInt.sml src/ffi/polyml/PolyMLFFIIntConvs-IntInfAsInt.sml
ln -sf PolyMLFFIIntConvs.sml src/ffi/polyml/PolyMLFFIIntConvs.sml
cc -DGIRAFFE_DEBUG -DGLIB_DISABLE_DEPRECATION_WARNINGS -fPIC -ggdb -std=c99 -Wall -Wno-deprecated-declarations -Werror -O2 -iquote "src" -c -o src/polyml/giraffe-cairo-1.0.o `pkg-config --cflags cairo-gobject` src/polyml/giraffe-cairo-1.0.c
src/polyml/giraffe-cairo-1.0.c:14:10: fatal error: 'cairo/cairo.h' file not found
#include <cairo/cairo.h>
         ^~~~~~~~~~~~~~~
1 error generated.
make: *** [src/polyml/giraffe-cairo-1.0.o] Error 1
```

I don't know what to make of it. I don't quite understand the error message of make.

**Comment by poldy** [comment]
Here are some minor changes I had to make to build some time ago: https://portal.mozz.us/gemini/republic.circumlunar.space/users/dbane/gemlog/patch-giraffe.diff.txt . Hopefully it will help.

**Comment by pclayton** [comment]
@poldy Thanks for linking to your patch.  The includes for cairo and Harfbuzz are fixed in https://github.com/giraffelibrary/giraffe/commit/d2e3c232a9589deca7fdf3168a3a01eae6bdd05f.

Without access to an arm Mac, it is hard to investigate the issues for other two changes in the patch.  Does `gio/gdesktopappinfo.h` not exist relative to the path in one of the -I flags returned by `pkg-config --cflags gio-unix-2.0`?  I had hoped that the GNU Make code for selecting the right variant of a file (using find-ceil and version-sort) would be portable.  One possibility is that a non-GNU `sort` utility is being used (which may give errors about the `-V` flag being invalid).

**Comment by poldy** [comment]
Here is some output from an ARM Mac:
```
$ pkg-config --cflags gio-unix-2.0
-I/opt/homebrew/Cellar/glib/2.74.5/include/gio-unix-2.0 -I/opt/homebrew/Cellar/glib/2.74.5/include -I/opt/homebrew/Cellar/glib/2.74.5/include/glib-2.0 -I/opt/homebrew/Cellar/glib/2.74.5/lib/glib-2.0/include -I/opt/homebrew/opt/gettext/include -I/opt/homebrew/Cellar/pcre2/10.42/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi
```
I checked any "gio" subdirectories of these that existed and found no gdesktopappinfo.h. This is after installing the "gdesktop-introspection" Homebrew package (and its dependencies). But since everything compiled for me, it looks like this #include file isn't required anyway? 

Also,
```
$ sort --version
2.3-Apple (154)
```
So I'd say your guess is correct.

I didn't find it difficult to make these small tweaks, but I understand as maintainer you want to make everything as friction-free as possible. If you want me to run any more checks just ask. I don't know if any of the free SCM hosting sites like github, gitlab, bitbucket, etc. offer ARM Mac CI pipelines, but that may also be more effort.

**Comment by pclayton** [comment]
> Here is some output from an ARM Mac:
> 
> ```
> $ pkg-config --cflags gio-unix-2.0
> -I/opt/homebrew/Cellar/glib/2.74.5/include/gio-unix-2.0 -I/opt/homebrew/Cellar/glib/2.74.5/include -I/opt/homebrew/Cellar/glib/2.74.5/include/glib-2.0 -I/opt/homebrew/Cellar/glib/2.74.5/lib/glib-2.0/include -I/opt/homebrew/opt/gettext/include -I/opt/homebrew/Cellar/pcre2/10.42/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi
> ```
> 
> I checked any "gio" subdirectories of these that existed and found no gdesktopappinfo.h. This is after installing the "gdesktop-introspection" Homebrew package (and its dependencies). But since everything compiled for me, it looks like this #include file isn't required anyway?

Correct, it is not required for Poly/ML but is for MLton (which uses wrapper C functions for passing array arguments, the merits of which is a whole other discussion).

> Also,
> 
> ```
> $ sort --version
> 2.3-Apple (154)
> ```
> 
> So I'd say your guess is correct.

The Makefile looks for `gnusort` and `gsort` first before falling back on `sort`, as Mac package managers may add a prefix to avoid conflicts.  If the prefixed versions are not available on the path then that would likely be the issue.

> I didn't find it difficult to make these small tweaks, but I understand as maintainer you want to make everything as friction-free as possible. If you want me to run any more checks just ask. I don't know if any of the free SCM hosting sites like github, gitlab, bitbucket, etc. offer ARM Mac CI pipelines, but that may also be more effort.

Thank you for offering.  I am making arrangements to test on an ARM Mac as that seems to be quite common now.
